### PR TITLE
Admins refactor

### PR DESCRIPTION
**Pull request by mumoc**
## Quick Info

Migrations? 👎

## What does this change?

- Removes unnecessary relationships from Supplier to Users.
- Updates abilities and roles; now we'll have a `supplier_admin` and `supplier_staff` roles

## Why are you changing that?

There were some relationships that weren't used at all; also, we found the need on having multiple admins for a given supplier and to define different role levels for them.

### CODE REVIEW

**Comment by kainio** [comment]
@mumoc The role is general, I mean it's applicable to all the marketplace not just for a particular supplier, is there a way to make the role specific for each supplier instead?

What I meant is someone can have a supplier_staff role for a supplier and can also have a supplier_admin role for another supplier at the same time.

Or maybe I misinterpreted the code.

**Comment by mumoc** [comment]
@kainio Let me explain what was the need:

Admin -> this is the main role, the marketer, the one that create the marketplace
Supplier Admin -> Is the one that get subscribed to the marketplace and have it's own products

With some feedback from some Suppliers, they asked to have someone else helping them in managing the store but didn't wanted them to have all the privileges. That's why we wanted to have a "Staff" role but we didn't define exactly what this role would do. 

What we think it should be is:

- Allow to admin inventory
- Allow to admin orders
- Restrict Supplier info
- Restrict Product information edit
- Restrict Pricing modification

Regarding your question: One same Admin cannot be assigned to different suppliers, if you have 2 suppliers and want to have admin and staff for those suppliers they would need to have different emails (users in the end)

- Supplier Alfa
   - Jose -> jose@alfa.com  role Admin
   - Jhon -> jhon@alfa.com role Staff

- Supplier Beta
   - Jhon -> jhon@beta.com role Admin
   - Juan -> juan@beta.com role Staff

File: spec/examples.txt [hunk 1]
```diff
@@ -0,0 +1,139 @@
+example_id                                                                      | status  | run_time        |
```
Comment (ccarruitero) [inline spec/examples.txt]: I think we should ignore this file

File: spec/models/spree/permission_sets/supplier/staff_ability_spec.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      it { expect(ability).to_not be_able_to :admin, other_resource }
+      it { expect(ability).to_not be_able_to :edit, other_resource }
+    end
+
+    context 'requested by suppliers user' do
+      let(:resource) {
+        product = create(:product)
+        product.add_supplier!(user.supplier)
+        product.reload
+      }
```
Comment (ccarruitero) [inline spec/models/spree/permission_sets/supplier/staff_ability_spec.rb]: This seems a bit weird for me. I think could be better something like
```suggestion
      let(:resource) { create(:product) }
      before do
        product.add_supplier!(user.supplier)
        product.reload
      end
```

Comment (mumoc) [reply spec/models/spree/permission_sets/supplier/staff_ability_spec.rb]: done ^^ thanks

File: spec/examples.txt (hunk 1 above)
Comment (mumoc) [reply spec/examples.txt]: done

File: spec/models/spree/permission_sets/supplier/staff_ability_spec.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+  context 'for StockItem' do
+    let(:resource) { Spree::StockItem }
+
+    context 'requested by another suppliers user' do
+      let(:resource) {
+        supplier = create(:supplier)
+        variant = create(:product).master
+        variant.product.add_supplier! supplier
+        supplier.stock_locations.first.stock_items.first
```
Comment (ccarruitero) [inline spec/models/spree/permission_sets/supplier/staff_ability_spec.rb]: I think we can refactor this like in 
https://github.com/magma-labs/solidus_marketplace/blob/d17243eaaa7ecf4b807deef03bbc5791e549fd21/spec/models/spree/permission_sets/supplier/staff_ability_spec.rb#L36-L41

File: spec/models/spree/permission_sets/supplier/staff_ability_spec.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      
+      it { expect(ability).to_not be_able_to :admin, resource }
+    end
+
+    context 'requested by suppliers user' do
+      let(:resource) {
+        variant = create(:product).master
+        variant.product.add_supplier! user.supplier
+        user.supplier.stock_locations.first.stock_items.first
+      }
```
Comment (ccarruitero) [inline spec/models/spree/permission_sets/supplier/staff_ability_spec.rb]: we can refactor the resource declaration here too.

Also we are declaring `variant` here and in other spec context too, we can move variant declaration on top of `StockItem` context and can reuse in all context.

File: spec/models/spree/permission_sets/supplier/staff_ability_spec.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  end
+
+  context 'for StockLocation' do
+    context 'requested by another suppliers user' do
+      let(:resource) {
+        supplier = create(:supplier)
+        variant = create(:product).master
+        variant.product.add_supplier! supplier
+        supplier.stock_locations.first
+      }
```
Comment (ccarruitero) [inline spec/models/spree/permission_sets/supplier/staff_ability_spec.rb]: same here

File: spec/models/spree/permission_sets/supplier/staff_ability_spec.rb [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+      it { expect(ability).to_not be_able_to :read, resource }
+    end
+
+    context 'requested by suppliers user' do
+      let(:resource) {
+        variant = create(:product).master
+        variant.product.add_supplier! user.supplier
+        user.supplier.stock_locations.first
+      }
```
Comment (ccarruitero) [inline spec/models/spree/permission_sets/supplier/staff_ability_spec.rb]: same here

**Review by ccarruitero — APPROVED** [review]

**Pull request merged**